llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -1548,7 +1548,16 @@ IRBuilder<>::InsertPoint OpenMPIRBuilder::createParallel(
   BasicBlock *CommonExit = nullptr;
   SetVector<Value *> Inputs, Outputs, SinkingCands, HoistingCands;
   Extractor.findAllocas(CEAC, SinkingCands, HoistingCands, CommonExit);
-  Extractor.findInputsOutputs(Inputs, Outputs, SinkingCands);
+
+  Extractor.findInputsOutputs(Inputs, Outputs, SinkingCands,
+                              /*CollectGlobalInputs=*/true);
+
+  Inputs.remove_if([&](Value *I) {
```

Comment:
Would it not be better to NOT add the global value in the first place, if it's an `OpenMPIRBuilder::Ident`? You already do `isa` in the code below, so doing a `GV = dyn_cast_if_present` instead and follow that with a `getValueType != OpenMPIRBuilder::Ident`?

Reply:
The problem with that is:
- `OpenMPIRBuilder::Ident` are created and maintained by the functions of the `OpenMPIRBuilder`. So I think it would be preferable to keep the knowledge of these structs within the builder.
- The `CodeExtractor` is not OpenMP-related and I think it would better not to leak any non-related OpenMP-details to it.

Of course it is better not to add something and then remove it. But I think it logically isolates things better this way.

Reply:
@Leporacanthicus is it fine with you merging this PR? Let me know if you do not agree with my reply above.